### PR TITLE
Implement url type in input component

### PR DESCRIPTION
**Pull request by JavierCenteno**

### CODE REVIEW

File: projects/lux/src/lib/input/input.component.ts [hunk 1]
```diff
@@ -253,6 +254,10 @@ export class InputComponent implements OnInit, ControlValueAccessor, Validator {
       result = result || {};
       result.email = { value, reason: 'Invalid email.' };
     }
+    if (this.type === 'url' && hasValue(value) && !isValidUrl(value)) {
+      result = result || {};
+      result.email = { value, reason: 'Invalid URL.' };
```
Comment (Foxandxss) [inline projects/lux/src/lib/input/input.component.ts]: email?

**Review by Foxandxss — CHANGES_REQUESTED** [review]

File: projects/lux/src/lib/input/input.component.ts (hunk 1 above)
Comment (JavierCenteno) [reply projects/lux/src/lib/input/input.component.ts]: Fixed: https://github.com/metadevpro/lux/pull/171/commits/baef477528547c9ab15410f039712070a4122fe9

File: projects/lux/src/lib/helperFns.ts [hunk 2]
```diff
@@ -16,6 +16,12 @@ export const isValidEmail = (value: string): boolean => {
   return re.test(String(value).toLowerCase().trim());
 };
 
+export const isValidUrl = (value: string): boolean => {
+  const re =
+    /^(https?:\/\/)?((([a-z\d]([a-z\d-]*[a-z\d])*)\.)+[a-z]{2,}|((\d{1,3}\.){3}\d{1,3}))(\:\d+)?(\/[-a-z\d%_.~+]*)*(\?[;&a-z\d%_.~+=-]*)?(\#[-a-z\d_]*)?$/i;
```
Comment (pjmolina) [inline projects/lux/src/lib/helperFns.ts]: Uris could use other non WWW protocols like:
- ftp://
- ws://
- wss://
- mongo://
- gopher://  ;-)
- etc.

Or are we constraining here only to WWW uris for a good reason?

Comment (JavierCenteno) [reply projects/lux/src/lib/helperFns.ts]: Fixed: https://github.com/metadevpro/lux/pull/171/commits/f04d951c8a680519ff62d0ba3d5aca82b07c6b39

File: projects/lux/src/lib/input/input.component.scss [hunk 3]
```diff
@@ -18,3 +23,23 @@
...
+  z-index: 1;
+}
+.clickable {
+  cursor: pointer;
+}
+.icon-external {
+  @extend .icon;
+  background: $icon-external;
+  background-size: 1rem;
+  height: 100%;
```
Comment (pjmolina) [inline projects/lux/src/lib/input/input.component.scss:45]: Add to center:
`background-position-y: 0.1rem;`

File: projects/lux/src/lib/input/input.component.html [hunk 4]
```diff
@@ -51,12 +51,18 @@
 <div *ngIf="isPermillage()" class="postfix rounded-right symbol" [ngClass]="className">
   <span>‰</span>
 </div>
+<div *ngIf="isUrl()" class="postfix rounded-right symbol clickable" [ngClass]="className">
```
Comment (pjmolina) [inline projects/lux/src/lib/input/input.component.html]: Show only if HAS value:   `*ngIf="isUrl() && !!value"`

**Review by pjmolina — CHANGES_REQUESTED** [review]
See comments

File: projects/lux/src/lib/input/input.component.html (hunk 4 above)
Comment (JavierCenteno) [reply projects/lux/src/lib/input/input.component.html]: Fixed:
https://github.com/metadevpro/lux/pull/171/commits/6e29a16039503955327a59e3fbed4ed3aae5737d
https://github.com/metadevpro/lux/pull/171/commits/bf36fd685a761721886349eb54edbaef5f65d196

File: projects/lux/src/lib/input/input.component.scss (hunk 3 above)
Comment (JavierCenteno) [reply projects/lux/src/lib/input/input.component.scss:45]: Added: https://github.com/metadevpro/lux/pull/171/commits/10b6d084bddcf09add57b33d34fa4d457bff1ff1

File: projects/lux/src/lib/input/input.component.ts [hunk 5]
```diff
@@ -66,13 +67,15 @@ export class InputComponent implements OnInit, ControlValueAccessor, Validator {
       required: 'Required field.',
       min: 'Minimum value is $min.',
       max: 'Maximum value is $max.',
-      email: 'Format should match example@example.com.'
+      email: 'Format should match example@example.com.',
+      url: 'Format should match example.com.'
```
Comment (pjmolina) [inline projects/lux/src/lib/input/input.component.ts]: Error message: We are going to require the protocol, right?
Then change to:  `url: 'Format should match a full URI: https://example.com.'`

Comment (JavierCenteno) [reply projects/lux/src/lib/input/input.component.ts]: Updated error message: https://github.com/metadevpro/lux/pull/171/commits/46d0b6f90168b069cf677e51ef03e4e7562dc55d
Can create a new method to check whether a URL is valid and includes the protocol if protocol is required

**Review by pjmolina — APPROVED** [review]
lgtm

**Pull request merged**